rust/src/client.rs

```diff
@@ -100,103 +104,91 @@ impl Client {
         }
     }
 
-    /// [`Client`] duties within a round
     pub async fn during_round(&mut self) -> Result<Task, ClientError> {
+        debug!(client_id = %self.id, "polling for new round parameters");
         loop {
-            if let Some(round_params_data) = self.handle.get_round_parameters().await {
-                if let Some(ref round_params) = round_params_data.round_parameters {
-                    if round_params.pk != self.coordinator_pk {
-                        // new round: save coordinator pk
-                        self.coordinator_pk = round_params.pk;
-                        debug!(client_id = %self.id, "computing sigs and checking task");
-                        let round_seed = round_params.seed.as_slice();
+            if let Some(params_outer) = self.proxy.get_params().await? {
```

Comment:
I think that in a separate ticket we should try to optimize this code block a bit. It would be nice if the processing of the current participant task stops as soon as the client has received new parameters. However, I don't think it's easy to do. Maybe it is not necessary after the tower integration.

Reply:
>It would be nice if the processing of the current participant task stops as soon as the client has received new parameters

agree that a lot probably changes soon, but not sure what you mean here?